tools/packaging/common/istio-iptables.sh

```diff
@@ -442,6 +436,12 @@ for gid in ${PROXY_GID}; do
   iptables -t nat -A ISTIO_OUTPUT -m owner --gid-owner "${gid}" -j RETURN
 done
 
+if [ -z "${DISABLE_REDIRECTION_ON_LOCAL_LOOPBACK-}" ]; then
+  # Redirect app calls back to itself via Envoy when using the service VIP or endpoint
+  # address, e.g. appN => Envoy (client) => Envoy (server) => appN.
+  iptables -t nat -A ISTIO_OUTPUT -o lo ! -d 127.0.0.1/32 -j ISTIO_REDIRECT
+fi
+
```

Comment:
This is more complicated than expect, changing the rules orders lead to 
traffic from appN => Envoy(outbound)=>appN, and this is why the e2e failed

Reply:
@rshriram I could not think of  a way to solve this.

Reply:
But it still work, right? the traffic still can go back to itself. its just don't go through envoy twice than before.

Reply:
I'm still confuse about why this ci failed, in my rough opinion, exchange the order of iptables won't affect the ci because the path that the traffic gothrough is the same, whether we change the order or not.

Reply:
>But it still work, right? the traffic still can go back to itself. its just don't go through envoy twice than before.

correct


> I'm still confuse about why this ci failed, in my rough opinion, exchange the order of iptables won't affect the ci because the path that the traffic gothrough is the same, whether we change the order or not.

I mean the e2e, if you look carefully, it will check the envoy logs to count some words to infer how many times request through envoy.

Reply:
Thanks, I find the code of e2e logCheck.

And I repeat the situation in my cluster, and I actually occurs when requests to SVC_IP:Port, and it has NO difference when using POD_IP:Port.

So, can we just modify the condition of logCheck to let it pass ? =.=

Reply:
How can you say that?

According to the current rules order, no matter calling by serviceip or podip, it is first intercepted and redirected to envoy, and then any traffic out of envoy will not be intercepted by envoy.

Reply:
Yes, that's absolute what happened, maybe its not a good idea~ I don't think too much.